tests/test_sparkkernelbase.py

```diff
@@ -309,7 +306,6 @@ def test_execute_throws_if_fatal_error_happens_for_execution():
         # Assertions
         assert kernel._fatal_error == message
         assert execute_cell_mock.call_count == 1
-        send_error_mock.assert_called_once_with(stream_content)
```

Comment:
This assertion is still needed. It's just that you would now verify with the ipython_display mock.
You might need to add `ipython_display=None` to SparkKernelBase constructor.